### PR TITLE
fix php pid/listen

### PR DESCRIPTION
**Pull request by plandolt**
This PR fixes the problem with creating or updating vm due to https://github.com/oerdnj/deb.sury.org/issues/1334.

### CODE REVIEW

File: vagrant-default.yml [hunk 1]
```diff
@@ -442,10 +442,12 @@ php:
                 xdebug.remote_enable: On
                 xdebug.remote_connect_back: On
         php_fpm:
+            global:
+                pid: /run/php/php7.2-fpm.pid
             default:
                 user: vagrant
                 group: vagrant
-                listen: /var/run/php/default.sock
+                listen: /run/php/php7.2-fpm.sock
```
Comment (dominikzogg) [inline vagrant-default.yml]: please do not add version specific paths

Comment (dominikzogg) [reply vagrant-default.yml]: and it needs to be aligned with line 315

Comment (dominikzogg) [reply vagrant-default.yml]: probably we have to adapt the `PIDFile=/run/php/php7.2-fpm.pid` in the systemd config, i don't think the socket change is needed

Comment (plandolt) [reply vagrant-default.yml]: I see.

Yes, the socket does not need to be changed.

One solution would also be just to remove the `PIDFile` from `/etc/systemd/system/multi-user.target.wants/php7.X-fpm.service` completely. Should I go this way with this PR?

Comment (dominikzogg) [reply vagrant-default.yml]: depends what is easier to achieve:

add would be similar as here: https://github.com/vagrant-php/ubuntu/blob/18.04/ansible/roles/general/tasks/main.yml#L59-L64

File: ansible/roles/php/tasks/main.yml [hunk 2]
```diff
@@ -34,6 +34,16 @@
     name: "php{{ php.version }}-fpm"
   when: role.nginx
 
+- name: fix php-fpm service definition regarding pid file
+  replace:
+    path: "/etc/systemd/system/multi-user.target.wants/php{{ php.version }}-fpm.service"
+    regexp: '^#?PIDFile\=(.*)'
+    replace: '#PIDFile=\1'
```
Comment (dominikzogg) [inline ansible/roles/php/tasks/main.yml:41]: when: role.nginx

File: ansible/roles/php/tasks/main.yml [hunk 3]
```diff
@@ -34,6 +34,16 @@
...
+
+- name: force systemd to reread configs
+  systemd:
+    daemon_reload: yes
```
Comment (dominikzogg) [inline ansible/roles/php/tasks/main.yml:46]: when: role.nginx

File: ansible/roles/php/tasks/main.yml (hunk 2 above)
Comment (plandolt) [reply ansible/roles/php/tasks/main.yml:41]: done

File: ansible/roles/php/tasks/main.yml (hunk 3 above)
Comment (plandolt) [reply ansible/roles/php/tasks/main.yml:46]: done

**Comment by dominikzogg** [comment]
will test if it works after a vagrant destroy

**Comment by dominikzogg** [comment]
offtopic: do you also have the issue, that you need to manually delete the certificates in certs/* since the last ssl changes in this repo ones you start after a `vagrant destroy`?

**Comment by plandolt** [comment]
> offtopic: do you also have the issue, that you need to manually delete the certificates in certs/* since the last ssl changes in this repo?

Yes, sort of. I just always provide certificates from the outside. But @brsart is aware and will give it a shot sometime.

**Comment by dominikzogg** [comment]
@scuben thanks for the effort

**Pull request merged**